core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -1224,7 +1224,7 @@ private[chisel3] object Builder extends LazyLogging {
     def default: State = State(
       currentModule = Builder.currentModule,
       whenStack = Nil,
-      blockStack = Nil,
+      blockStack = Builder.blockStack,
```

Comment:
This was a little surprising to me, but the old behavior is to leave the block stack in place and let the new module see it. This seems wrong, however, this is how it works.

I'm envisioning a larger refactor of this which makes a lot of this more straightforward. I.e., a module should have a single block created inside it. It shouldn't see the parent block stack. I haven't followed through on the ramifications of this change.

Reply:
Yep! It's weird for sure.  Thanks for straightening this all out, I know I'd be happy to see the sort of "order" you're proposing brought to these bits.  But what you have already sure helps clean things up a lot, woohoo!

I can point you to further related bits (like Placeholder -> apply methods) if you'd like re:ramifications, but also you seem to be cruising so maybe best to just keep on 👍 .  Nothing that won't fall out, I don't think.